### PR TITLE
added time frequencies to regex for T

### PR DESCRIPTION
**Pull request by kthyng**
My goal in this is to expand what is matched to "count" as time when `guess_coord_axis` is run to include other time frequencies that might result from for example a groupby call. With the change in this PR, the following is now possible

image

This avoids needing to add attributes after the calculation 

`ds.temp.isel(xi_rho=300,s_rho=-1).cf.groupby("T.hour").mean(skipna=False)`

### CODE REVIEW

**Comment by dcherian** [comment]
Thanks! Any interest in adding a test? We don't test these regexes or coordinate criteria yet so it is not a big deal.

File: cf_xarray/accessor.py [hunk 1]
```diff
@@ -109,7 +109,7 @@
 
 #: regular expressions for guess_coord_axis
 regex = {
-    "time": "time[0-9]*",
+    "time": "time[0-9]*|min|hour|day|week|month|year",
```
Comment (dcherian) [inline cf_xarray/accessor.py:112]: Can you add the attributes listed here: https://xarray.pydata.org/en/stable/generated/xarray.core.accessor_dt.DatetimeAccessor.html and https://xarray.pydata.org/en/stable/generated/xarray.core.accessor_dt.TimedeltaAccessor.html? (the ones that make sense anyway like `dayofyear`)

**Comment by kthyng** [comment]
I hope to get back to this this week.

**Comment by dcherian** [comment]
Sounds good. We could also detect `ocean_time` or any `*time*` variable.

**Comment by dcherian** [comment]
Merging. We can upgrade this functionality later.

**Pull request merged**